content/blog/nats-operator-and-service-accounts.md

```diff
@@ -0,0 +1,337 @@
+that uses the ServiceAccounts integration by toggling the
+`enableServiceAccounts` flag as part of the `NatsCluster` spec.  In order to
+support configuration changes on-the-fly for the cluster, we toggle
+`enableConfigReload` as well.
+
+```yaml
+---
+apiVersion: nats.io/v1alpha2
+kind: NatsCluster
+metadata:
+  name: example-nats
+spec:
+  size: 3
+  version: "1.2.0"
+  pod:
+    enableConfigReload: true
+  auth:
+    enableServiceAccounts: true
+```
+
+In the [nats-operator](https://github.com/nats-io/nats-operator/) repository there is a [complete example](https://github.com/nats-io/nats-operator/blob/84d06026c70412e172335e673b1dffad33fe66c1/example/example-cluster-service-accounts.yaml) of creating a
+NATS cluster with service accounts defined for it, so let's deploy
+that and take a look at the result:
+
+```sh
+$ kubectl -n nats-io apply -f https://raw.githubusercontent.com/nats-io/nats-operator/master/example/example-cluster-service-accounts.yaml
+natscluster.nats.io/example-nats created
+serviceaccount/nats-admin-user created
+serviceaccount/nats-user created
+nnatsservicerole.nats.io/nats-user unchanged
+natsservicerole.nats.io/nats-admin-user unchanged
+pod/nats-user-pod created
+pod/nats-admin-user-pod created
+```
+
+As it can be seen from the output, the command above has created a couple
+of service accounts, one for a `nats-user` and another one for a
+`nats-admin-user`, as well as a couple of `natsservicerole` objects with
+the same names.  The `NatsServiceRole` is a new
+`CustomResourceDefinition` that the NATS Operator creates to represent
+a `ServiceAccount` that will be used by a NATS cluster.
+
+For example, in the case of the `nats-user`, a `NatsServiceRole` is
+defined as below so that the `example-nats` cluster that was created by the
+operator sets PUB/SUB permissions for that account:
+
+```yaml
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: nats-user
+---
+apiVersion: nats.io/v1alpha2
+kind: NatsServiceRole
+metadata:
+  name: nats-user
+  namespace: nats-io
+
+  # Specifies which NATS cluster will be using this role.
+  labels:
+    nats_cluster: example-nats
+spec:
+  permissions:
+    publish: ["foo.*", "foo.bar.quux"]
+    subscribe: ["foo.bar", "greetings", "hello.world"]
+```
+
+The NATS Operator will then detect that there is a mapping between a
+NatsServiceRole and a ServiceAccount and then issue a new token where
+the audience is set to be the operated NATS cluster service and storing
+it into a secret:
+
+```sh
+kubectl -n nats-io get secret nats-user-example-nats-bound-token  -o yaml
+apiVersion: v1
+data:
+  token: ZXlKaGJHY2lPaUpTVXp...
+kind: Secret
+metadata:
+  creationTimestamp: 2018-07-20T00:48:25Z
+  labels:
+    app: nats
+    nats_cluster: example-nats
+  name: nats-user-example-nats-bound-token
+  namespace: nats-io
+```
+
+Now a Pod in order to use the `nats-user` account, can then mount the secret
+as a volume to be able to use that token to authenticate:
+
```

Comment:
In order to use the `nats-user` account, a pod can now mount the secret as a volume to use that token to authenticate: